Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -12835,8 +12848,324 @@ def pan_os_delete_application_group_command(args):
+    
+    # calculate largest unique id for each log type query
+    new_largest_id = max({entry.get('seqno', '') for entry in incident_entries})
+
+    # convert incident entries to incident context and filter any empty incidents if exists
+    parsed_incidents: List[Dict[str, Any]] = [incident_entry_to_incident_context(incident_entry) for incident_entry in incident_entries]
+    filtered_parsed_incidents = list(filter(lambda incident: incident, parsed_incidents))
+    demisto.debug(f'{len(filtered_parsed_incidents)} parsed incidents returned from parse_incident_entries function')
+    if filtered_parsed_incidents:
+        demisto.debug(f"parsed incidents unique ID list: {[incident.get('name','') for incident in filtered_parsed_incidents]}")
+    return new_largest_id, new_fetch_datetime, filtered_parsed_incidents
+
+
+def incident_entry_to_incident_context(incident_entry: Dict[str, Any]) -> Dict[str, Any]:
+    """convert raw incident entry to basic cortex incident format.
+
+    Args:
+        incident_entry (Dict[str, Any]): raw incident entry represented by a dictionary
+
+    Returns:
+        dict[str,any]: context formatted incident entry represented by a dictionary
+    """
+    occurred = incident_entry.get('time_generated', "")
+    occurred_datetime = dateparser.parse(occurred, settings={'TIMEZONE': 'UTC'})
+    if occurred_datetime:
+        incident_context = {
+            'name': incident_entry.get('seqno'),
+            'occurred': occurred_datetime.strftime(DATE_FORMAT),
+            'rawJSON': json.dumps(incident_entry),
+            'type': incident_entry.get('type')
+        }
+        return incident_context
+    else:
+        return {}
+
+
+def get_fetch_start_datetime_dict(last_fetch_dict: Dict[str, str],
+                                  first_fetch: str, queries_dict: Optional[Dict[str, str]]) -> Dict[str, datetime]:
+    """calculate fetch start time for each log type query.
+    - if last fetch time already exists for a log type, it will not be changed (only converted to datetime object).
+    - if last fetch time does not exist for the log_type, it will be changed into first_fetch parameter (and converted to datetime object).
+    - example: {'log_name':'2022-12-18T05:58:17'} --> {'log_name': datetime.datetime(2022, 12, 18, 5, 58, 17)}
+
+    Args:
+        last_fetch_dict (Dict[str,str]): last fetch dictionary
+        first_fetch (str): first fetch parameter
+        queries_dict (Optional[Dict[str, str]]): queries per log type dictionary
+
+    Returns:
+        Dict[str,datetime]: log_type:datetime pairs dictionary
+    """
+    fetch_start_datetime_dict = {}
+    first_fetch_parsed = dateparser.parse(first_fetch, settings={'TIMEZONE': 'UTC'})
+
+    # add new log types to last_fetch_dict
+    if queries_dict:
+        for log_type in queries_dict.keys():
+            if (log_type not in last_fetch_dict) and (log_type != "All"):
+                last_fetch_dict[log_type] = ''
```

Comment:
why do we need it?

Reply:
this loop adds new 'log type' keys to last_fetch_dict, this is needed for the next code segment that handles this.

Reply:
this loop adds new 'log type' keys to last_fetch_dict, this is needed for the next code segment that handles this.